### PR TITLE
[YUNIKORN-2145] refactor: ApplicationSummary into its own file

### PR DESCRIPTION
**Pull request by brandboat**
### What is this PR for?
ApplicationSummary is a type on its own which should not be included in the generic application file. Move the type and related functions into its own application_summary.go file. Move the related tests out too.

### What type of PR is it?
* [x] - Improvement

### Todos
N/A

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2145

### How should this be tested?
covered by unit tests

### Screenshots (if appropriate)
N/A

### Questions:
N/A

### CODE REVIEW

File: pkg/scheduler/objects/application_summary.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		as.SubmissionTime.UnixMilli(),
+		as.StartTime.UnixMilli(),
+		as.FinishTime.UnixMilli(),
+		as.User,
+		as.Queue,
+		as.State,
+		as.RmID,
+		as.ResourceUsage.TrackedResourceMap,
+		as.PreemptedResource.TrackedResourceMap,
+		as.PlaceholderResource.TrackedResourceMap)
```
Comment (brandboat) [inline pkg/scheduler/objects/application_summary.go:60]: The `TrackedResource` to String method has been implemented in https://github.com/apache/yunikorn-core/pull/713. Currently, it has not been merged into the master branch. Therefore, we will continue to use TrackedResourceMap here.

Comment (pbacsko) [reply pkg/scheduler/objects/application_summary.go:60]: I'll ping the owner of #713.

File: pkg/scheduler/objects/application_summary_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+package objects
+
+import (
+	"testing"
+
+	"gotest.tools/v3/assert"
+)
+
+func TestNilApplicationSummaryToString(t *testing.T) {
```
Comment (pbacsko) [inline pkg/scheduler/objects/application_summary_test.go:27]: Is this necessary? I haven't seen any unit test which tests sth like that... I don't think we need this.

File: pkg/scheduler/objects/application_summary.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	RmID                string
+	ResourceUsage       *resources.TrackedResource
+	PreemptedResource   *resources.TrackedResource
+	PlaceholderResource *resources.TrackedResource
+}
+
+func (as *ApplicationSummary) String() string {
+	if as == nil {
+		return "nil ApplicationSummary"
+	}
```
Comment (pbacsko) [inline pkg/scheduler/objects/application_summary.go:47]: Do we need this "nil" check?

File: pkg/scheduler/objects/application_summary.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+package objects
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/apache/yunikorn-core/pkg/common/resources"
+	"github.com/apache/yunikorn-core/pkg/log"
+	"go.uber.org/zap"
```
Comment (pbacsko) [inline pkg/scheduler/objects/application_summary.go:27]: Fix the imports.

**Review by pbacsko — CHANGES_REQUESTED** [review]
See comments. I don't think we need the nil check & test.

File: pkg/scheduler/objects/application_summary.go (hunk 3 above)
Comment (brandboat) [reply pkg/scheduler/objects/application_summary.go:47]: For now, we don't need this nil check. I'll remove the nil check and test code in the next commit, many thanks.

**Pull request closed without merging**